File: ts/src/push-to-function/fs/boot/foo-sfl-boot.ts
/*
 * Copyright © 2022 Paulo Villela. All rights reserved.
 * Use of this source code is governed by the MIT license
 * that can be found in the LICENSE file.
 */

import { AppCfgInfo} from "../../config/app-cfg";
import { fooSflC, FooSflCfgInfo, FooSflT } from "../foo-sfl";
import { barBfBoot } from "./bar-bf-boot";
import { mutableLifted } from "../../fwk/mutable-lifted";

function adapter(appCfgInfo: AppCfgInfo): FooSflCfgInfo {
  return appCfgInfo
}

export const fooSflCfgAdapter = mutableLifted(adapter)

export function fooSflBoot(appCfgSrc: () => AppCfgInfo): FooSflT {
  const cfgSrc = {
    get: fooSflCfgAdapter(appCfgSrc),
    bar: barBfBoot(appCfgSrc)
  }
  return fooSflC(cfgSrc)
}

File: go/push-to-function/fs/boot/foo-sfl-boot.go
/*
 * Copyright © 2022 Paulo Villela. All rights reserved.
 * Use of this source code is governed by the MIT license
 * that can be found in the LICENSE file.
 */

package boot

import (
	"github.com/pvillela/module-config/go/push-to-function/config"
	"github.com/pvillela/module-config/go/push-to-function/fs"
	"github.com/pvillela/module-config/go/push-to-function/fwk"
)

func fooSflCfgAdapter(appCfgInfo config.AppCfgInfo) fs.FooSflCfgInfo {
	return fs.FooSflCfgInfo{
		X: appCfgInfo.X,
	}
}

var FooSflCfgAdapter = fwk.LiftToNullary(fooSflCfgAdapter)

func FooSflBoot(appCfgSrc config.AppCfgSrc) fs.FooSflT {
	return fs.FooSflC(fs.FooSflCfgSrc{
		Get:   FooSflCfgAdapter(appCfgSrc),
		BarBf: BarBfBoot(appCfgSrc),
	})
}
